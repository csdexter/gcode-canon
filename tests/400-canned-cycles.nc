(testing canned cycles)
(machine setup)
M05 S600
M09
M23
M49
M69

(turn servos ON)
M17

(control setup)
G15
G17
G23
G40
G49
G50
G69


(in mm)
G21

(following code taken from the official RS274NGC spec)
G17 M3 S1234 F600 (START IN XY-PLANE)
G81 X3 Y4 R0.2 Z-1.1
X1 Y0 R0 G91 L3 (THREE MORE G81’S AN INCH APART)
Y-2 R0.1 (ONE MORE G81)
G82 G90 X4 Y5 R0.2 Z-1.1 P0.6
X2 Z-3.0 (ONE MORE G82)
G91 X-2 Y2 R0 L4 (FOUR MORE G82’S)
G83 G90 X5 Y6 R0.2 Z-1.1 Q0.21
G84 X6 Y7 R0.2 Z-1.1
G85 X7 Y8 R0.2 Z-1.1
G86 X8 Y9 R0.2 Z-1.1 P902.61
G87 X9 Y10 R0.2 Z-1.1 I0.231 J-0 K-3
G91 X1 R0.2 Z-1.1 I0.231 J-0 K-3
G88 X10 Y11 R0.2 Z-1.1 P0.3333
G89 X11 Y12 R0.2 Z-1.1 P1.272
M4 (RUN SPINDLE COUNTERCLOCKWISE)
G86 X8 Y9 R0.2 Z-1.1 P902.61
G87 X9 Y10 R0.2 Z-1.1 I0.231 J-0 K-3
G88 X10 Y11 R0.2 Z-1.1 P0.3333

M2 (THE END)
