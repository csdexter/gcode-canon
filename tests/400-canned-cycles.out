MSG: WAR: Machine servos activated!
MSG: STA: Scanning input for programs (O words)
MSG: WAR: Machine servos activated!
MPOS,0.00,0.00,0.20
MPOS,3.00,4.00,0.20
MPOS,3.00,4.00,-1.10
MPOS,3.00,4.00,0.20
MPOS,3.00,4.00,0.00
MPOS,4.00,4.00,0.00
MPOS,4.00,4.00,-1.10
MPOS,5.00,4.00,-1.10
MPOS,5.00,4.00,0.00
MPOS,5.00,4.00,-1.10
MPOS,5.00,4.00,0.00
MPOS,6.00,4.00,0.00
MPOS,6.00,4.00,1.10
MPOS,6.00,4.00,0.00
MPOS,6.00,4.00,1.10
MPOS,6.00,4.00,1.20
MPOS,6.00,2.00,1.20
MPOS,6.00,2.00,0.10
MPOS,6.00,2.00,0.00
MPOS,0.00,-2.00,0.20
MPOS,4.00,5.00,0.20
MPOS,4.00,5.00,-1.10
MPOS,4.00,5.00,0.20
MPOS,4.00,5.00,-1.10
MPOS,4.00,5.00,0.20
MPOS,2.00,5.00,0.20
MPOS,2.00,5.00,-3.00
MPOS,2.00,5.00,0.20
MPOS,2.00,5.00,-1.10
MPOS,0.00,7.00,-1.10
MPOS,0.00,7.00,-4.10
MPOS,-2.00,9.00,-4.10
MPOS,-2.00,9.00,-1.10
MPOS,-2.00,9.00,-4.10
MPOS,-2.00,9.00,-1.10
MPOS,-4.00,11.00,-1.10
MPOS,-4.00,11.00,1.90
MPOS,-4.00,11.00,-1.10
MPOS,-4.00,11.00,1.90
MPOS,-6.00,13.00,1.90
MPOS,-6.00,13.00,4.90
MPOS,-6.00,13.00,1.90
MPOS,-6.00,13.00,4.90
MPOS,-2.00,2.00,0.20
MPOS,5.00,6.00,0.20
MPOS,5.00,6.00,-0.01
MPOS,5.00,6.00,0.20
MPOS,5.00,6.00,0.10
MPOS,5.00,6.00,-0.01
MPOS,5.00,6.00,-0.22
MPOS,5.00,6.00,0.20
MPOS,5.00,6.00,-0.11
MPOS,5.00,6.00,-0.22
MPOS,5.00,6.00,-0.43
MPOS,5.00,6.00,0.20
MPOS,5.00,6.00,-0.32
MPOS,5.00,6.00,-0.43
MPOS,5.00,6.00,-0.64
MPOS,5.00,6.00,0.20
MPOS,5.00,6.00,-0.53
MPOS,5.00,6.00,-0.64
MPOS,5.00,6.00,-0.85
MPOS,5.00,6.00,0.20
MPOS,5.00,6.00,-0.74
MPOS,5.00,6.00,-0.85
MPOS,5.00,6.00,-0.89
MPOS,5.00,6.00,0.20
MPOS,5.00,6.00,-3.00
MPOS,5.00,6.00,0.20
MPOS,6.00,7.00,0.20
MPOS,6.00,7.00,-1.10
MPOS,6.00,7.00,0.20
MPOS,6.00,7.00,-1.10
MPOS,6.00,7.00,0.20
MPOS,7.00,8.00,0.20
MPOS,7.00,8.00,-1.10
MPOS,7.00,8.00,0.20
MPOS,7.00,8.00,-1.10
MPOS,7.00,8.00,0.20
MPOS,8.00,9.00,0.20
MPOS,8.00,9.00,-1.10
MPOS,8.00,9.00,0.20
MPOS,8.00,9.00,-1.10
MPOS,8.00,9.00,0.20
MPOS,9.00,10.00,0.20
MPOS,9.23,10.00,0.20
MPOS,0.23,0.00,-1.10
MPOS,9.00,10.00,-1.10
MPOS,9.00,10.00,-3.00
MPOS,9.00,10.00,-1.10
MPOS,9.23,10.00,-1.10
MPOS,0.23,0.00,0.20
MPOS,9.00,10.00,0.20
MPOS,9.00,10.00,-1.10
MPOS,9.00,10.00,-0.90
MPOS,10.00,10.00,-0.90
MPOS,10.23,10.00,-0.90
MPOS,10.23,10.00,-2.00
MPOS,10.00,10.00,-2.00
MPOS,10.00,10.00,-5.00
MPOS,10.00,10.00,-2.00
MPOS,10.23,10.00,-2.00
MPOS,10.00,10.00,-2.00
MPOS,10.00,10.00,-2.20
MPOS,10.00,10.00,-2.00
MPOS,20.00,21.00,-2.00
MPOS,20.00,21.00,-3.10
MPOS,20.00,21.00,-3.30
MPOS,20.00,21.00,-3.10
MPOS,31.00,33.00,-3.10
MPOS,31.00,33.00,-4.20
MPOS,31.00,33.00,-4.40
MPOS,31.00,33.00,-4.20
MPOS,39.00,42.00,-4.20
MPOS,39.00,42.00,-5.30
MPOS,39.00,42.00,-5.50
MPOS,39.00,42.00,-5.30
MPOS,48.00,52.00,-5.30
MPOS,48.23,52.00,-5.30
MPOS,48.23,52.00,-6.40
MPOS,48.00,52.00,-6.40
MPOS,48.00,52.00,-9.40
MPOS,48.00,52.00,-6.40
MPOS,48.23,52.00,-6.40
MPOS,48.00,52.00,-6.40
MPOS,48.00,52.00,-6.60
MPOS,48.00,52.00,-6.40
MPOS,58.00,63.00,-6.40
MPOS,58.00,63.00,-7.50
MPOS,58.00,63.00,-7.70
